### DISCUSSION TITLE
p12 public key cannot be installed on Apple device anymore

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#1554] [Q&A] [answered]
Roadwarrior mode

I have just generated new key pairs on my server. The client's public key (.pem/der/cer) was converted into p12 format by OpenSSL on a Linux server. Possible errors in the phase of Export passwords have been checked repeatedly and I have tried many times including no password set. However, I failed to add the p12 certificate to the profile via Apple Configurator 2. Even without a password set, the Mac told me the input password was wrong. With passwords set, I also got feedback saying the password was wrong.

This issue never happened before, but it turns up after macOS has upgraded. I suspect recent macOS updates blocked the self-signed p12 certificates (by computers other than the Mac itself.)

Any help would be much appreciated.

**Comment by tobiasbrunner** [comment, marked as answer]
Did you follow the advice in the docs when OpenSSL 3 is involved (i.e. using the `-legacy` option, it's mentioned [here](https://docs.strongswan.org/docs/5.9/interop/appleIkev2Profile.html#_known_issues) and [here](https://docs.strongswan.org/docs/5.9/pki/pkiQuickstart.html#_pkcs12_container))?

**Reply by TomHsiung** [reply]
Oh. Hmmm, the problem I described in the above post has been fixed with your suggestion (-legacy). Thanks.
